src/Twig/Extension/TemplateRegistryExtension.php

```diff
@@ -11,59 +11,82 @@
 
 namespace Sonata\AdminBundle\Twig\Extension;
 
-use Sonata\AdminBundle\Admin\AdminInterface;
-use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
 final class TemplateRegistryExtension extends AbstractExtension
 {
     /**
-     * @var Pool
+     * @var TemplateRegistryInterface
      */
-    private $pool;
+    private $globalTemplateRegistry;
 
-    public function __construct(Pool $pool)
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(TemplateRegistryInterface $globalTemplateRegistry, ContainerInterface $container)
     {
-        $this->pool = $pool;
+        $this->globalTemplateRegistry = $globalTemplateRegistry;
+        $this->container = $container;
     }
 
     public function getFunctions()
     {
         return [
             new TwigFunction('get_admin_template', [$this, 'getAdminTemplate']),
-            new TwigFunction('get_admin_pool_template', [$this, 'getPoolTemplate']),
+            new TwigFunction('get_global_template', [$this, 'getGlobalTemplate']),
+
+            // NEXT MAJOR: Remove this line
+            new TwigFunction('get_admin_pool_template', [$this, 'getGlobalTemplate'], ['deprecated' => true]),
```

Comment:
Shouldnt this be getPoolTemplate as before?

Reply:
I renamed the "pool" templates to "global" templates. They're the same list, just named differently.
They used to be stored inside the `Pool` class, that's why I named the function "get_admin_pool_template" at first. But "global" is a better name.

Reply:
If you are restoring the old function, maybe you should restore the old call for this deprecated function. To preserve behavior

Reply:
Alright.